public/controller/data_functions.php

```diff
@@ -135,44 +144,134 @@ function upload_homework($username, $semester, $course, $assignment_id, $homewor
+        if(!file_exists($assignment_path)) {
+          display_error("Folder ".$assignment_path." from previous submission does not exist.");
+          return;
+        }
 
+        // check if folder for this user exists
+        $user_path = $assignment_path."/".$username;
+        if(!file_exists($assignment_path)) {
+          display_error("Folder ".$assignment_path." from previous submission does not exist.");
+          return;
+        }
 
-    $filename="";
-    $extension="";
+        //Find the previous homework version number
+        $previous_version = 1;
+        while (file_exists($user_path."/".$previous_version)) {
+          // FIXME: Replace with symlink
+          $previous_version++;
+        }
+        $previous_version -= 1;
 
-    if ($svn_checkout==false) {
-      if ($homework_file["size"] / 1024 > $max_size || !is_valid_zip_size($homework_file["tmp_name"], $max_size)) {
-        return array("error"=>"", "message"=>"File uploaded is too large.  Maximum size is ".$max_size." kb. Uploaded file was ".$homework_file["size"] / 1024 ." kb.");
-      }
+        // No previous submission, should not have entered this if statement
+        if($previous_version <= 0) {
+          display_error("No submission found. There should not be any files kept from previous submission.");
+          return;
+        }
 
-      $filename = explode(".", $homework_file["name"]);
-      $extension = end($filename);
+        $previous_version_path = $user_path."/".$previous_version;
+
+        // check if folders for parts exist
+        $previous_part_path = [];
+        // ===========================================================================================
+        // Uncomment the if-else statement if want single-parted hw to be put under the version folder
+        // ===========================================================================================
+        //if($num_parts > 1) {
+          // check folder for multi-part homework
+          for($n=1; $n <= $num_parts; $n++) {
+            $previous_part_path[$n] = $previous_version_path."/part".$n;
+            if (!file_exists($previous_part_path[$n])) {
+              display_error("Folder ".$previous_part_path[$n]." does not exist.");
+              return;
+            }
+          }
+        // }
+        // else {
+        //   $previous_part_path[$num_parts] = $previous_version_path;
+        // }
+
+        // check if files from previous version exist
+        for($n=1; $n<=$num_parts; $n++) {
+          if(isset($previous_files[$n])) {
+            for($i=0; $i<count($previous_files[$n]); $i++) {
+              $filename = $previous_part_path[$n]."/".$previous_files[$n][$i];
+              if (!file_exists($filename)) {
+                display_error("File ".$filename." does not exist.");
+                return;
+              }
+            }
+          }
+        }
+      }
     }
 
 
+    $max_size = 50;
```

Comment:
This should be set to the same default max submission size as config.h which is 100000 (50 bytes is pretty much always going to be too small anyway).